src/main/java/seedu/address/model/person/Person.java

```diff
@@ -77,13 +103,15 @@ public boolean equals(Object other) {
         return otherPerson.getName().equals(getName())
                 && otherPerson.getPhone().equals(getPhone())
                 && otherPerson.getEmail().equals(getEmail())
-                && otherPerson.getAddress().equals(getAddress());
+                && otherPerson.getAddress().equals(getAddress())
+                && otherPerson.getMoneyOwed().equals(getMoneyOwed())
+                && otherPerson.getMoneyPaid().equals(getMoneyPaid());
```

Comment:
correct me im wrong, should this not be added ? because if we have eg.  Alex Yeoh 91234567 with money paid $5 in addressbook. Then, after that if we add another person called Alex Yeoh 91234567 (money paid default to 0), there should be a duplicate error. But if we include this in equals() method, i don't think the addressbook will throw a duplicate error right?

Reply:
I think this `equals()` method in `Person` class differs from the `isSamePerson()` by establishing a stronger relationship that all fields of the `Person` instance should be equal to each other. From my understanding, this `equals()` method can be used in testing to ensure all the fields match and it would make sense to include the comparisons here.

Also this is also consistent with the comments of this function:

> /**
>      * Returns true if both persons have the same identity and data fields.
>      * This defines a stronger notion of equality between two persons.
>      */
>